### PR TITLE
[WIP] Added the GuzzleHttp adapter

### PR DESCRIPTION
**Pull request by Ninir**
This adds the guzzle Adapter.
Currently working on fixing tests.

Should this be opened against `next_version`?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,11 +25,12 @@
         }
     },
     "require" : {
-        "php"                : ">=5.3.3"
+        "php"                : ">=5.4"
```
Comment (Ninir) [inline composer.json]: As GuzzleHttp v4+ requires PHP 5.4, is this ok?
It will break previous applications...

**Comment by Ninir** [comment]
By the way, I'm having issues with `fopen(path/to/file.jpg): failed to open stream: No such file or directory`
How can I mock this? :/

File: composer.json (hunk 1 above)
Comment (Baachi) [inline composer.json]: Is it not needed, as the adapter is optional. :)

Comment (Ninir) [inline composer.json]: Reverted, thanks @Baachi :)
Do you have an idea regarding the fopen issue? ^^

**Comment by Baachi** [comment]
@Ninir Guzzle have a MockSubsciber, its the easiest way.
http://guzzle.readthedocs.org/en/latest/testing.html

**Comment by Ninir** [comment]
@Baachi The MockSubscriber is meant to queue responses or exceptions. Is there something I am missing?

**Comment by Baachi** [comment]
Yes, its the easies/best way to unit test it. There is also a [`MockHandler`](https://github.com/guzzle/RingPHP/blob/master/src/Client/MockHandler.php) but i never used it.

**Comment by henrikbjorn** [comment]
Thanks for your work on this :)

**Pull request merged**